katcp/resource_client.py

```diff
@@ -55,6 +55,23 @@ def _transform(f):
     future.add_done_callback(_transform)
     return new_future
 
+def until_all(futures):
+    """Run multiple operations in parallel but only raise a single TimeoutError.
+
+    This returns a yieldable object that will contain the results of `futures`
+    in an equivalent container (list or dict). However, it only raises the
+    first TimeoutError and ignores any further timeouts. This makes it useful
+    to wait on the results of multiple operations with the same timeout, as the
+    first occurrence of a timeout means that any unresolved futures will also
+    necessarily time out, leading to a rash of duplicate exceptions otherwise.
+
+    The function name implies kinship with :func:`katcp.core.until_any`.
+    The primary difference is that `until_any` takes an explicit timeout while
+    this function relies on the individual futures to have their own timeouts.
+
```

Comment:
Sure. I should tone down that word 'primary'. I assumed the names were suggestive enough of the meaning of the functions but wanted to point out a difference in their APIs. `until_any` takes an explicit `timeout` parameter while `until_all` assumes that the futures are already wrapped in timeouts themselves.